crates/typst-syntax/src/ast.rs

```diff
@@ -80,6 +80,40 @@ impl<'a> Markup<'a> {
     }
 }
 
+node! {
+    DecoratorName
+}
+
+impl<'a> DecoratorName<'a> {
+    /// Get the decorator name.
+    pub fn get(self) -> &'a EcoString {
+        self.0.text()
+    }
+
+    /// Get the decorator name as a string slice.
+    pub fn as_str(self) -> &'a str {
+        self.get()
+    }
+}
+
+node! {
+    /// A decorator: `/! allow("warning")`.
+    Decorator
```

Comment:
As discussed, I prefer the naming "annotation". We also discussed the syntax `// @allow warning`, but haven't 100% settled yet.

Reply:
The syntax has been updated, though we may want to confirm it through further discussion.